projects/cashmere/src/lib/slide-toggle/slide-toggle.component.ts

```diff
@@ -0,0 +1,165 @@
+        return this._componentId || this._uniqueId;
+    }
+
+    set id(idVal: string) {
+        this._componentId = idVal ? idVal : this._uniqueId;
+    }
+
+    /** An optional label positioned outside of the slide toggle. Use `labelPosition` to set which side. */
+    @Input() sideLabel = '';
+
+    /** The text inside the toggle. Choose from: `none`, `on` (ON/OFF), `true` (TRUE/FALSE), `yes` (YES/NO). *Defaults to `on`**/
+    @Input()
+    get insideLabel(): string {
+        return this._insideLabel;
+    }
+
+    set insideLabel(val: string) {
+        validateLabelType(val);
+        this._insideLabel = val;
+    }
+
+    /** Whether the individual toggle is disabled. */
+    @Input()
+    get disabled(): boolean {
+        return this._disabled;
+    }
+
+    set disabled(val: boolean) {
+        this._disabled = parseBooleanAttribute(val);
+    }
+
+    /** Sets whether this is a required form element */
+    @Input()
+    get required(): boolean {
+        return this._isRequired;
+    }
+
+    set required( requiredVal: boolean | string ) {
+        this._isRequired = parseBooleanAttribute(requiredVal);
+    }
+
+    /** Sets the size of slide toggle. Choose from: `'sm' | 'md' | 'lg' |`. *Defaults to `sm`.**/
+    @Input() buttonSize: buttonSize = buttonSize.small;
+
+    /** Sets the style of the slide toggle. Choose from: `'blue' | 'purple' | 'green' |`. *Defaults to `blue`.**/
+    @Input() buttonStyle: buttonStyle = buttonStyle.blue;
+
+    /** Sets which side of slide toggle the outside label is positioned on. Choose from: `'left' | 'right' |`. *Defaults to `left`.**/
+    @Input() labelPosition: labelPosition = labelPosition.left;
+
+    /** Sets the on/off state of the toggle. Supports two-way binding. */
+    @Input()
+    get buttonState(): boolean {
+        return this._buttonState;
+    }
+
+    set buttonState( val: boolean ) {
+        this._buttonState = parseBooleanAttribute(val);
+        this.onChange(this._buttonState);
+        this.buttonStateChanged.emit(this._buttonState);
+    }
+
+    /** Event fired with boolean value when the toggle state is changed. */
+    @Output() buttonStateChanged = new EventEmitter<boolean>();
+
+    constructor(
+        @Optional() _parentForm: NgForm,
+        @Optional() _parentFormGroup: FormGroupDirective,
+        @Optional() @Self() public _ngControl: NgControl
+    ) {
+        super();
+
+        this._form = _parentForm || _parentFormGroup;
+        if (this._ngControl != null) {
+            this._ngControl.valueAccessor = this;
+        }
+    }
+
+    writeValue(value: unknown): void {
+        this.buttonState = !!value;
+    }
+
+    public onChange: (value: unknown) => void = () => undefined;
+    public onTouch: () => unknown = () => undefined;
+    public registerOnChange(fn: (value: unknown) => void): void {
+        this.onChange = fn;
+    }
+    public registerOnTouched(fn: () => unknown): void {
+        this.onTouch = fn;
+    }
+
+    ngDoCheck(): void {
```

Comment:
I mentioned to @corykon that this implementation of `ngDoCheck` gives me heartburn, because a half-dozen of our form components do it.  It was done this way several years ago because there wasn't a good way to subscribe to form changes, but there _has_ to be a better way to do this.  On big forms you could have tons of components all calling doCheck several times a second 😬.  I've implemented it here as with other components in Cashmere, but I'll put an issue in for it.